src/rust-crypto/RoomEncryptor.ts

```diff
@@ -105,7 +108,15 @@ export class RoomEncryptor {
         const rustEncryptionSettings = new EncryptionSettings();
         /* FIXME historyVisibility, rotation, etc */
 
-        const shareMessages = await this.olmMachine.shareRoomKey(
+        const isRoomBlacklistUnverifiedDevices = this.room.getBlacklistUnverifiedDevices();
+        rustEncryptionSettings.onlyAllowTrustedDevices =
+            // When isRoomBlacklistUnverifiedDevices === null, the global settings should be used
+            // See Room#getBlacklistUnverifiedDevices
+            isRoomBlacklistUnverifiedDevices === null
+                ? globalBlacklistUnverifiedDevices
+                : isRoomBlacklistUnverifiedDevices;
```

Comment:
isn't this the same as:

```suggestion
            isRoomBlacklistUnverifiedDevices ?? globalBlacklistUnverifiedDevices;
```

?

Reply:
(consider inlining `isRoomBlacklistUnverifiedDevices` as well)